lib/jsdom/browser/Window.js

```diff
@@ -50,7 +50,7 @@ function Window(options) {
   // vm initialization is defered until script processing is activated (in level1/core)
   this._globalProxy = this;
 
-  this.__timers = [];
+  this.__timers = {};
```

Comment:
Nit: use Object.create(null) for these sorts of things, here and when doing stopAllTimers.